provider/kubernetes/kubernetes.go

```diff
@@ -104,13 +104,13 @@ func (p *Provider) Provide(configurationChan chan<- types.ConfigMessage, pool *s
 					case <-stop:
 						return nil
 					case event := <-eventsChan:
-						log.Debugf("Received event from kubernetes %+v", event)
+						log.Debugf("Received Kubernetes event kind %T", event)
```

Comment:
Does k8s provide some debugging functionality for the events? For example do they have an id which I could use to check the full event details via `kubectl`?

Reply:
Good question.

What we actually receive here over the channel is a message that gets triggered by the informer watch receiving updates on the observed objects (i.e., one of Ingresses, Services, Endpoints, and Secrets). I just verified locally that apparently updates to the objects of interest for us are not reflected by Kubernetes events: For an example application deployed, I don't see events other than those related to the container lifecycle (scheduled, created, scaled, etc.)

So overall, I don't think there's much we can leverage here. I won't completely rule out though I might have missed something important.

Reply:
Ok, thanks for the explanation.